### DISCUSSION TITLE
nginx-botsearch jail seen as malicious?

### DISCUSSION THREAD

**Discussion by Sinbad95** [fail2ban/fail2ban#3123] [Q&A]
Hi All,

I have recently started using HomeAssistant after a long break and am starting to setup the services. I also run malware protection on my network which has started reporting HomeAssistant trying to access a malicious URL. After checking various logs I find that my router had an entry for 79.138.10.109 at the correct time, which is being reported as malicious on several lists. The site [https://www.abuseipdb.com/] has some reports suggesting it is related to nginx-botsearch jail.

The fact that it has been blocked on my network is not an issue as I can easily unblock it but wanted to bring this to your attention in case this is related to you as it could affect others and if it is not doing anything malicious, you can see if you can get it removed from the blacklists.

**Comment by sebres** [comment]
> HomeAssistant trying to access a malicious URL
> it is not doing anything malicious

That are a bit contradicting statements, don't they? :)

> it is related to nginx-botsearch jail

Well, by default filter `nginx-botsearch` is searching for bots defined with `block` and other parameters in:
https://github.com/fail2ban/fail2ban/blob/5093ff89561a96aa4424432f22c752c5bdd7e0f6/config/filter.d/botsearch-common.conf#L6-L14

which may be not "optimal" for your purposes (due to crossing with some service/URI listing there).

Also note that this values are defaults and it can (or even must) be overwritten within your .local file(s).

Anyway, neither it is obvious what exactly was wrong in your case without to consider the log-line causing a failure (provide an example), nor it is possible to predict any constellation which could cause a misleading finding.

The usage of such filters is not advisable, at least unless you know what you do or you cannot guarantee that such URIs will be invoked from your services. 
Instead of usage of this jail, one could either configure web-server (nginx) to prohibit certain URIs/prefixes (return 403 and log it to some file) and ban them with a regexp finding 403rd responses, or ban evildoers doing too many requests causing 404 (with corresponding large values of `maxretry`/`findtime`) possibly with an exception list ignoring some URIs may occur regularly for legitimate users (e. g. because some pages have a dead links/references).

> but wanted to bring this to your attention in case this is related to you as it could affect others and if it is not doing anything malicious

I don't think it is somehow related to us, since fail2ban is just a tool and as any other tool should be properly configured.
It searching and banning that what generates failures related to your own configuration... With other words you decide what is malicious and what isn't.
